modules/porous_flow/src/materials/PorousFlow1PhaseP.C

```diff
@@ -64,30 +63,28 @@ PorousFlow1PhasePTempl<is_ad>::computeQpProperties()
 
   buildQpPPSS();
 
-  if (!is_ad)
-  {
-    const auto ds = _pc_uo.dSaturation(MetaPhysicL::raw_value(_porepressure_var[_qp]));
+  const auto ds = _pc_uo.dSaturation(_porepressure_var[_qp]);
 
-    if (!_nodal_material)
-    {
-      (*_gradp_qp)[_qp][0] = _gradp_qp_var[_qp];
-      (*_grads_qp)[_qp][0] = ds * _gradp_qp_var[_qp];
-    }
+  if (!_nodal_material)
+  {
+    (*_gradp_qp)[_qp][0] = _gradp_qp_var[_qp];
+    (*_grads_qp)[_qp][0] = ds * _gradp_qp_var[_qp];
+  }
 
     // _porepressure is only dependent on _porepressure, and its derivative is 1
-    if (_dictator.isPorousFlowVariable(_porepressure_varnum))
+  if (!is_ad && _dictator.isPorousFlowVariable(_porepressure_varnum))
+  {
+    // _porepressure is a PorousFlow variable
+    (*_dporepressure_dvar)[_qp][0][_p_var_num] = 1.0;
+    (*_dsaturation_dvar)[_qp][0][_p_var_num] = MetaPhysicL::raw_value(ds);
+
+    if (!_nodal_material)
     {
-      // _porepressure is a PorousFlow variable
-      (*_dporepressure_dvar)[_qp][0][_p_var_num] = 1.0;
-      (*_dsaturation_dvar)[_qp][0][_p_var_num] = ds;
-      if (!_nodal_material)
-      {
-        (*_dgradp_qp_dgradv)[_qp][0][_p_var_num] = 1.0;
-        (*_dgrads_qp_dgradv)[_qp][0][_p_var_num] = ds;
-        (*_dgrads_qp_dv)[_qp][0][_p_var_num] =
-            _pc_uo.d2Saturation(MetaPhysicL::raw_value(_porepressure_var[_qp])) *
-            _gradp_qp_var[_qp];
-      }
+      (*_dgradp_qp_dgradv)[_qp][0][_p_var_num] = 1.0;
+      (*_dgrads_qp_dgradv)[_qp][0][_p_var_num] = MetaPhysicL::raw_value(ds);
+      (*_dgrads_qp_dv)[_qp][0][_p_var_num] =
+          _pc_uo.d2Saturation(MetaPhysicL::raw_value(_porepressure_var[_qp])) *
+          MetaPhysicL::raw_value(_gradp_qp_var[_qp]);
```

Comment:
I was getting compile time errors without the `raw_value`'s, but looking into `constexp` it looks like it might avoid these. I'll give it a try. Thanks!

Reply:
Yep, this worked!